app/models/assignment.rb

```diff
@@ -1464,4 +1464,68 @@ def same_membership_as_csv_row?(row, existing_grouping)
     end
   end
 
+  # Reformatted logic for download_assignment_list controller method
+  # Returns an output file for controller to handle.
+  def self.get_assignment_list(file_format, default_fields)
+    assignments = self.all
+
+    case file_format
+    when 'yml'
+      map = {}
+      map[:assignments] = []
+      assignments.map do |assignment|
+        m = {}
+        default_fields.length.times do |i|
```

Comment:
Use `default_fields.map` to loop over elements rather than indexes.

Also `map` *returns* a list, so you can actually do `map[:assignments] = default_fields.map ...`, and remove the `<< m` part in the loop body.